strawberryfields/backends/bosonicbackend/backend.py

```diff
@@ -724,8 +730,20 @@ def state(self, modes=None, peaks=None, **kwargs):
         Returns:
             BosonicState: state description
         """
+        if isinstance(modes, int):
+            modes = [modes]
+
         if modes is None:
-            modes = list(range(len(self.get_modes())))
+            modes = self.get_modes()
+
+        mode_names = ["q[{}]".format(i) for i in modes]
+
+        if len(modes) == 0:
```

Comment:
why should not an error be raised if `len(modes)==0`?

Reply:
I made this change to accommodate the test_modes.py test, where one of the tests is if the user deletes all modes. In that case, the state object is still created but just with empty arrays. If it were up to me an error would be raised if a used tries to delete all modes, but it seems the other backends have been done this way.